File: tiq-e/others/2/main.ts
import {performance} from "perf_hooks";

function hammingDistance(x: number, y: number): number {
    let ans = 0
    for (let i = 0; i < 32; i++) {
        let xBit = x & 1
        let yBit = y & 1

        ans += xBit !== yBit ? 1 : 0

        x >>= 1
        y >>= 1
    }
    return ans
}

const x = 2;
const y = 4;

const start = performance.now();
const result = hammingDistance(x, y);
const end = performance.now();
console.log("Duration:", end - start);
console.log("Result:", result);

File: tiq-e/others/2/main.go
package main

import (
	"fmt"
	"time"
)

func hammingDistance(x int, y int) int {
	nCount := 0
	for r := x ^ y; r != 0; r >>= 1 {
		fmt.Println(r & 1)
		if r&1 == 1 {
			nCount++
		}
	}
	return nCount
}

func main() {
	x := 2
	y := 4

	start := time.Now()
	result := hammingDistance(x, y)
	end := time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", result)
}
